### DISCUSSION TITLE
Question about syntax

### DISCUSSION THREAD

**Discussion by JPR-source** [jspsych/jsPsych#905] [Q&A] [answered]
Hello,

Could someone explain the syntax: 'jsPsych.pluginAPI.getkeyboardResponse'

Thanks

**Comment by jodeleeuw** [comment, marked as answer]
Hi @JPR-source,

Can you elaborate a little bit? Is there something you are hoping to know beyond what's in the documentation? 

https://www.jspsych.org/core_library/jspsych-pluginAPI/#jspsychpluginapigetkeyboardresponse

**Reply by JPR-source** [reply]
I'm not very experienced with programming and javascript so just trying to understand the chain of nested parts here, what they are, and where they come from. As in this is referencing 'getkeyboardresponse' which is a? inside 'PluginAPI' which is a? , inside 'jsPsych' which is a?
Are these things loaded in using '<script src="jspsych/jspsych.js"></script>' which is a function that creates 'jsPsych' and all these things contained within?
ps. sorry if this is the wrong place for such a question

**Reply by vijaymarupudi** [reply]
`<script src="jspsych/jspsych.js"></script>` is a HTML tag that fetches the Javascript and then executes it.

`jsPsych` is an [`Object`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) defined in `jspsych.js`.

Objects are key-value pairs of data. The keys are usually strings.  For the key `pluginAPI` in the object `jsPsych`, the value is another object. In this object, for the key `getKeyboardResponse`, the value is a [`function`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function).

**Reply by jodeleeuw** [reply]
@JPR-source totally the right place for this question -- just wasn't sure what you were asking! please feel free to keep asking questions here.

**Reply by JPR-source** [reply]
Thanks, that's really helpful! Apologies for the initial vagueness!
